dask/dataframe/tests/test_utils_dataframe.py

```diff
@@ -521,3 +523,30 @@ def test_assert_eq_sorts():
     assert_eq(df1, df2_r, check_index=False)
     with pytest.raises(AssertionError):
         assert_eq(df1, df2_r)
+
+
+def test_assert_eq_scheduler():
+    using_custom_scheduler = False
+
+    def custom_scheduler(*args, **kwargs):
+        nonlocal using_custom_scheduler
+        try:
+            using_custom_scheduler = True
+            return get_sync(*args, **kwargs)
+        finally:
+            using_custom_scheduler = False
+
+    def check_custom_scheduler(part: pd.DataFrame) -> pd.DataFrame:
+        assert using_custom_scheduler
+        return part + 1
+
+    df = pd.DataFrame({"x": [1, 2, 3, 4]})
+    ddf = dd.from_pandas(df, npartitions=2)
+    ddf2 = ddf.map_partitions(check_custom_scheduler, meta=ddf)
+
+    with pytest.raises(AssertionError):
+        assert_eq(ddf2, ddf2)
+
+    assert_eq(ddf2, ddf2, scheduler=custom_scheduler)
```

Comment:
I think this is close to what we want, but it's not clear to me that this is actually using `custom_scheduler`. For example, if we just added a `scheduler=`  keyword to `assert_eq` and then did nothing with it, this line would still pass. If `custom_scheduler` incremented some counter when it was called and we then `assert`ed that the counter has been incremented the expected number of times after calling `assert_eq` it would be more straightforward to see that `custom_scheduler` is actually being used (there certainly could be other approaches too -- I'm just throwing one possibility out there)

Reply:
The point here was that if you _weren't_ using `custom_scheduler`, the computation would fail (as just tested above). I could also have the `custom_scheduler` function increment a counter every time it's called, but that seems kind of redundant to me.

Reply:
I checked it against itself just because it seemed like an easier way to verify that both sides of the `assert_eq` were respecting the `scheduler=` logic. Otherwise, besides parametrizing the test to flip them around, it would be possible to have a future regression where one of the calls inside `assert_eq` forgot to pass the argument through.